services/src/contexts/migrations/schema_info.rs

```diff
@@ -0,0 +1,401 @@
+    referential_constraints,
+    join = "NATURAL JOIN information_schema.table_constraints",
+    "constraint_schema",
+    "table_name ASC, unique_constraint_name ASC, constraint_name ASC",
+    (table_name, String),
+    (constraint_name, String),
+    (unique_constraint_name, String),
+    (match_option, String),
+    (update_rule, String),
+    (delete_rule, String),
+    (constraint_type, String),
+    (is_deferrable, String),
+    (initially_deferred, String)
+);
+
+schema_info_table!(
+    SchemaTableKeyConstraint,
+    key_column_usage,
+    join = "NATURAL JOIN information_schema.table_constraints",
+    "table_schema",
+    "table_name ASC, constraint_name ASC, ordinal_position ASC, column_name ASC",
+    (constraint_name, String),
+    (table_name, String),
+    (column_name, String),
+    (constraint_type, String),
+    (position_in_unique_constraint, Option<i32>),
+    (is_deferrable, String),
+    (initially_deferred, String)
+);
+
+schema_info_table!(
+    SchemaUserDefinedTypes,
+    user_defined_types,
+    "user_defined_type_schema",
+    "user_defined_type_name ASC",
+    (user_defined_type_name, String)
+);
+
+schema_info_table!(
+    SchemaParameters,
+    parameters,
+    "specific_schema",
+    "specific_name, ordinal_position ASC",
+    (specific_name, String),
+    (ordinal_position, i32),
+    (parameter_mode, String),
+    (parameter_name, String),
+    (data_type, String),
+    (character_maximum_length, Option<i32>),
+    (character_octet_length, Option<i32>),
+    (numeric_precision, Option<i32>),
+    (numeric_precision_radix, Option<i32>),
+    (numeric_scale, Option<i32>),
+    (datetime_precision, Option<i32>),
+    (interval_type, Option<String>),
+    (interval_precision, Option<i32>),
+    (udt_name, String),
+    (parameter_default, Option<String>)
+);
+
+schema_info_table!(
+    SchemaDomain,
+    domains,
+    "domain_schema",
+    "domain_name, udt_name ASC",
+    (domain_name, String),
+    (data_type, String),
+    (character_maximum_length, Option<i32>),
+    (character_octet_length, Option<i32>),
+    (numeric_precision, Option<i32>),
+    (numeric_precision_radix, Option<i32>),
+    (numeric_scale, Option<i32>),
+    (datetime_precision, Option<i32>),
+    (interval_type, Option<String>),
+    (interval_precision, Option<i32>),
+    (domain_default, Option<String>),
+    (udt_name, String)
+);
+
+async fn tables(transaction: &Transaction<'_>, schema_name: &str) -> Result<Vec<String>> {
+    Ok(transaction
+        .query(
+            "
+            SELECT
+                table_name
+            FROM
+                information_schema.tables
+            WHERE
+                table_schema = $1
+            ORDER BY table_name ASC",
+            &[&schema_name],
+        )
+        .await?
+        .iter()
+        .map(|row| row.get("table_name"))
+        .collect::<Vec<String>>())
+}
+
+pub struct AssertSchemaEqConfig {
+    pub has_views: bool,
+}
+
+pub async fn assert_schema_eq(
```

Comment:
seems to work, but it's not really clear to me that we actually check everything here.

Maybe it would also be a good idea to write 2-3 sentences about the approach, because the code is not really that easy to understand.

Reply:
[04905a8](https://github.com/geo-engine/geoengine/pull/942/commits/04905a83d155787fb7031fdbf222bc59e208ec1a)